mixer/adapter/rbac/rbacStore.go

```diff
@@ -304,8 +424,52 @@ func valueMatch(a interface{}, b string) bool {
 	return false
 }
 
-// Check if all properties defined for the subject are satisfied by the properties from the request.
-func checkSubject(properties map[string]interface{}, subject map[string]string) bool {
+// checkSubjectsMatch returns true if rule subjects and instance subjects are matched. Otherwise
+// returns false.
+func checkSubjectsMatch(subject *rbacproto.Subject, instance *authorization.Instance) bool {
```

Comment:
The name could just be `checkSubjectMatch`, we're only checking one `subject` here.

Another question, the `instance` is the user provided value, the `subject` is the current RBAC policy. so can we change the logic in this function to be based on `subject`, not the `instance`?

So it could be something like:
```
func checkSubjectsMatch(subject *rbacproto.Subject, instance *authorization.Instance) bool {
matched := false
if subject.GetUser() != "" {
  if instance.Subject.User doesn't match subject.GetUser() {
    return false
  }
  matched = true
}
if subject.GetGroup() != "" {
  if instance.Subject.Group doesn't match subject.GetGroup() {
    return false
  }
  matched = true
}
if subject.GetProperties() != "" {
  if instance.Subject.Properties doesn't match subject.GetProperties() {
    return false
  }
  matched = true
}
return matched
}
```

Reply:
You are right. Fixed.